test/unit/account_test.rb

```diff
@@ -864,4 +864,23 @@ def test_smart_destroy_master
     assert master_account.reload
     refute master_account.reload.scheduled_for_deletion?
   end
+
+  test 'bought_cinstances_count when buying a cinstance' do
+    account = FactoryBot.create(:simple_buyer, provider_account: master_account)
```

Comment:
```suggestion
    account = FactoryBot.create(:simple_provider, provider_account: master_account)
```
Because a tenant is a buyer of master which is what you are doing 🤔